packages/sitecore-jss-nextjs/src/editing/vercel-editng-data-cache.ts

```diff
@@ -0,0 +1,60 @@
+import { VercelKV, createClient } from '@vercel/kv';
+import { EditingDataCache } from './editing-data-cache';
+import { EditingData } from './editing-data';
+import { debug } from '@sitecore-jss/sitecore-jss';
+
+/**
+ * Implementation of editing cache for Vercel deployments
+ * Uses Vercel KV database and client to store data
+ * Set TTL for cache data in constructor (default: 60 seconds)
+ */
+export class VercelEditingDataCache implements EditingDataCache {
+  protected redisCache: VercelKV;
+  private ttl;
+  
+  /**
+   * @param {string} redisUrl KV endpoint URL. Usually stored in process.env.KV_REST_API_URL
+   * @param {string} redisToken KV endpoint tokem. Usually stored in process.env.KV_REST_API_TOKEN
+   * @param {string} [entryTtlSeconds] TTL of cache entries in second. Default: 60 
+   */
+  constructor(redisUrl: string | undefined, redisToken: string | undefined, entryTtlSeconds = 60) {
+    if (!redisUrl || !redisToken) {
+      throw Error(
+        'API URL or token are missing, ensure you have set the KV or Upstash storage correctly.'
+      );
+    }
+    this.ttl = entryTtlSeconds;
+    this.redisCache = createClient({
+      url: redisUrl,
+      token: redisToken,
+    });
+  }
+
+  set(key: string, editingData: EditingData): Promise<void> {
+    debug.editing(`Putting editing data for ${key} into redis storage...`);
+    return new Promise<void>((resolve, reject) => {
+      this.redisCache
+        .set(key, JSON.stringify(editingData))
+        .then(() => {
+          this.redisCache.expire(key, this.ttl).then(() => {
```

Comment:
a get call will follow a set almost always - but it's not guaranteed. Is it still better to invalidate on get instead of set?
might be better to do both to ensure there's no orphan data left

Reply:
I like the idea of doing it in both (something long like 60 in the set, 0 in the get). I think about the editing cache more like a stack data structure with push/pop so in that sense the get(pop) is the place to do immediate cleanup.